runtime/doc/options.txt

```diff
@@ -2186,6 +2186,26 @@ A jump table for the options with a short description can be found at |Q_op|.
 	This option is reset when the 'paste' option is set and restored when
 	the 'paste' option is reset.
 
+				*'filechangenotify'* *'fcnotify'*
+'filechangenotify' 'fcnotify'	string (default "off")
+				local to buffer
+	Watch for changes to a file. How to react for the changes depends on
+	the value of the option. The option can take a list of following values:
+
+		off		Don't react to changes happening to a file.
+				Can't be set with anyother of the following
+				values.
+		autoread	Notify about the changes to a file and prompt
+				the user if there are changes in the current
+				buffer. If there are no changes reload the
+				buffer without prompts.
+				In the absence of this value the user will be
+				prompted for every change.
+		watcher		Use filesystem notifications for watching a
+				file. Might not work on all filesystems.
+		onfocus		Use aucmds to check for changes to a file on
+				the events `FocusGained`
```

Comment:
These are low-level details. No reason to expose them as permanent options. The implementation might change/improve, and then these options are just noise.

Reply:
I was thinking of changing the option and having it accept a single value. The name of the backend we want to use for watching the files. OnFocus, fswatch or something else if we later support it. Having autoread here is redundant since we already have an autoread option that controls the behaviour of `checktime`. 

So we could simply have `fcnotify=watcher` or `fcnotify=onfocus` or `fcnotify=off`.